Update geolocation.js polyfill

anssiko opened the pull request:
This is an attempt to update the geolocation.js polyfill started by @tobie to better match the latest Generic Sensor API.

It seems at least https://w3c.github.io/sensors/examples/realtime-positioning/ example works after my patches. I haven't attempted to run the test suite.

Related, here's an early draft of the Geolocation Sensor spec:

https://anssiko.github.io/geolocation-sensor/

PTAL @rwaldron

anssiko commented:
I fixed couple of bugs that were introduced by my previous changes.

PTAL @alexshalamov @pozdnyakov

alexshalamov commented on examples/realtime-positioning/proposal.js:
```diff
@@ -1,13 +1,9 @@
-sensor.onchange = function(event) {
-    var coords = [event.reading.latitude, event.reading.longitude];
-    updateMap(null, coords, event.reading.accuracy);
+sensor.onreading = function(event) {
```
Maybe remove event, since it doesn't carry any useful information.

anssiko commented:
@alexshalamov, thanks for the swift review. I addressed your comments in https://github.com/w3c/sensors/pull/268/commits/3f41e3a9da1799a7545c9d2d61b02249ff333108

kenchris commented on examples/realtime-positioning/proposal.js:
```diff
@@ -1,13 +1,9 @@
     updateMap(error);
 };
-sensor.onstatechange = function(e) {
-    console.log("STATE:", this.state)
```
I see you are removing this, but <this> would be wrong here... why not move them all to array methods?

anssiko replied:
PRs for further enhancements welcome :-)

rwaldron commented on examples/realtime-positioning/proposal.js:
```diff
@@ -1,13 +1,9 @@
-    var coords = [event.reading.latitude, event.reading.longitude];
-    updateMap(null, coords, event.reading.accuracy);
+sensor.onreading = function() {
+    var coords = [sensor.latitude, sensor.longitude];
```
This is so much nicer :)
